File: backend/app/schemas/academic.py
import uuid
from datetime import time
from pydantic import BaseModel

class CourseResponse(BaseModel):
    id: uuid.UUID
    course_code: str
    title: str
    credit_hours: float
    type: str

class ClassScheduleResponse(BaseModel):
    id: uuid.UUID
    course_code: str
    course_title: str
    day_of_week: str
    start_time: str
    end_time: str
    room_number: str
    instructor_name: str
    is_lab: bool

File: frontend/src/types/academic.ts
export interface Course {
  id: string;
  course_code: string;
  title: string;
  credit_hours: number;
  type: string;
}

export interface Semester {
  id: number;
  title: string;
  is_active: boolean;
}

export interface ClassSchedule {
  id: string;
  course_code: string;
  course_title: string;
  day_of_week: string;
  start_time: string;
  end_time: string;
  room_number: string;
  instructor_name: string;
  is_lab: boolean;
}
